### ISSUE TITLE
Wrong return value of toDecimalString()

### ISSUE THREAD

**Issue by mightymercado** [rwpenney/spfpm#4]
```
>>> FXnum(+0.5, family = FXfamily(n_bits=1,n_intbits=3)).toBinaryString()
'000.1'
FXnum(+0.5, family = FXfamily(n_bits=1,n_intbits=3)).toDecimalString()
'0.'
```

The decimal string should be 0.5

**Comment by rwpenney** [comment]
Thanks for spotting this. Finding a perfect solution to the problem of converting a binary fixed-point number into decimal that works well for all bit-depths is proving somewhat subtle!

I've adjusted the FXnum.toDecimalString() method so that it should have a better estimate of the maximum number of decimal digits that it can safely print, and so that it does work for the case you highlighted. There's also now an optional 'precision' argument that you can use to expand or limit the number of printed digits when necessary. (More unit-tests have been added to capture this case for future reference.)

**Comment by rwpenney** [comment]
I believe this was fixed as of commit 4894b1a4.
